lib/middleware/route-handlers.js

```diff
@@ -65,7 +107,9 @@ var setup = function(sourceFiles, cb) {
 
 };
 
-module.exports = function(sourceFiles, cb) {
+module.exports = function(options, cb) {
+    var sourceFiles = options.sourceFiles;
+    autoOptions = options.autoOptions;
```

Comment:
Yep agree totally about moving the parsing out, that's definitely something we have on our "roadmap".

We want to eventually be able to modify routes at runtime.